File: find_the_parity_outlier.rb
# An Array contains at least 3 array items
# Array is comprised entirely of odd integers or even integers
# Array will always containe a single outlier
# Return the array's outliner

#####
#
# sepearate an array to odd and even numbers
# find an array that only contains one array item
# return the value of the array
def find_outlier(integers)
  integers.partition { |a| a.even? }.find { |a| a.count == 1 }.first
end

File: find_the_parity_outlier.js
const findOutlier = (integers) => {
    let odd = integers.filter(integer => ](Math.abs(integer%2) == 1));
    let even = integers.filter(integer => (Math.abs(integer%2) == 0));
    return odd.length == 1 ? odd[0] : even[0]
  }